lib/resty/dns/resolver.lua

```diff
@@ -978,5 +1013,23 @@ function _M.reverse_query(self, addr)
                       {qtype = self.TYPE_PTR})
 end
 
+function _M.socks_cleanup(self, sock_typs)
+    if type(sock_typs) ~= 'table' then
+        return nil, "sock_typs must be an array"
+    end
+
+    local opts = { sock_typs[1], sock_typs[2] }
```

Comment:
sock_typs is an array, why create a new table below?

Reply:
only takes the first two even if we get more than expected